garmin_txt.cc

```diff
@@ -834,17 +826,19 @@ garmin_txt_write()
 
   if (waypoints > 0) {
     wpt_a_ct = 0;
-    wpt_a = (const Waypoint**)xcalloc(waypoints, sizeof(*wpt_a));
+    wpt_a = new const Waypoint*[waypoints]{};
     waypt_disp_all(enum_waypt_cb);
     route_disp_all(nullptr, nullptr, enum_waypt_cb);
-    qsort(wpt_a, waypoints, sizeof(*wpt_a), sort_waypt_cb);
+    auto sort_waypt_lambda = [](const Waypoint* wa, const Waypoint* wb)->bool {
+      return wa->shortname.compare(wb->shortname, Qt::CaseInsensitive) < 0;
```

Comment:
This doesn't have to preserve the subsort on index because we're assuming that nobody actually cares about the order and that stable_sort() will keep the nuanced sort cases under wraps for cases like our tests, right?

It seems weird to have this in garmin_txt, but I'm not at all sure how garmin_txt is used in the real world or if it is at all.

Reply:
duplicate (not garmin_txt) didn't preserve the final sort key of index, but the new sort should work exactly the same as the old sort.  We are NOT assuming nobody actually cares about the order, the behavior is equivalent and matches the preexisting comment in duplicate.cc L84-115.   In this case stable_sort was not used for test case portability.  Stable sorting algorithms preserve the relative order of equal elements.  Adding the index as the final sort key is a method to get a stable sort using qsort.  Stable behavior is part of the std::stable_sort contract: "The order of equivalent elements is guaranteed to be preserved."

This isn't the first conversion from qsort to std:stable_sort that has allowed us to simplify the sorted objects and comparison function by dropping the index.  However, there is at least one usage of stable_sort elsewhere that was done solely for portability of test cases.

Reply:
I don't know why garmin_txt sorts waypoints, I just re-implemented the previous behavior.  I do wonder if garmin_txt, and several other formats are ever used.  So far you  have done a remarkable job of deprecating formats without complaint!